### PR TITLE
RATIS-1820. Update apache parent pom version and other versions.

### PR DESCRIPTION
**Pull request by szetszwo**
See https://issues.apache.org/jira/browse/RATIS-1820

### CODE REVIEW

**Review by adoroszlai — COMMENTED** [review]
Thanks @szetszwo for working on this.

`ratis-examples` is missing from the source tarball created by `build.sh`, that's why _compile (11)_ check is failing.

File: pom.xml [hunk 1]
```diff
@@ -207,7 +185,7 @@
     <maven.min.version>3.3.9</maven.min.version>
 
     <!-- Contains all shaded thirdparty dependencies -->
-    <ratis.thirdparty.version>1.0.3</ratis.thirdparty.version>
+    <ratis.thirdparty.version>1.0.4</ratis.thirdparty.version>
```
[the remark this answers is not in the document]
Comment (szetszwo) [reply pom.xml]: @adoroszlai , thanks for the reminder.  Let #862 do the thirdparty update.

**Comment by szetszwo** [comment]
Tried to update Apache parent pom to version 29 (and 28, 27, 26).  However, the assembled src tarball somehow is missing the ratis-examples module.  Apache parent pom to version 25 does not have such problem.

**Comment by szetszwo** [comment]
@adoroszlai , thanks a lot of reviewing this!

**Pull request merged**